homeassistant/components/inels/config_flow.py

```diff
@@ -0,0 +1,180 @@
+"""Config flow for iNELS."""
+
+from __future__ import annotations
+
+from typing import Any
+
+from inelsmqtt import InelsMqtt
+from inelsmqtt.const import MQTT_TRANSPORT
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.config_entries import ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_PASSWORD, CONF_PORT, CONF_USERNAME
+from homeassistant.core import HomeAssistant, callback
+
+from .const import DOMAIN, TITLE
+
+CONNECTION_TIMEOUT = 5
+
+
+class FlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle of iNELS config flow."""
+
+    VERSION = 1
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: config_entries.ConfigEntry,
+    ) -> InelsOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return InelsOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
```

Comment:
why?

Reply:
In a typical Home Assistant setup, the Mosquitto MQTT broker is often installed as an add-on. Given this common configuration, it's practical to use a single broker connection. The underlying inelsmqtt library is designed to work with one broker instance.

Reply:
The term *discovery* is ambiguous in this particular context.

Do you mean: 

1. discovering an MQTT broker running as a Home Assistant add-on? Leveraging the async_step_hassio & ask the user to confirm the broker configuration.
2. MQTT discovery where devices/integrations publish directly to topics "homeassistant/+/+/config".

Reply:
Good point, i think neither.

Checkout `drop_connect`. They have an mqtt topic in the manifest.json, and if they recognise a message on that topic, it will auto discover the integration. Using async step mqtt in the config flow